src/main/java/com/example/carservice/service/CarServiceImpl.java

```diff
@@ -11,12 +11,26 @@ public class CarServiceImpl implements CarService {
 
     private final CarRepository repository;
 
-    public CarServiceImpl(CarRepository repository) {
+
+    public CarServiceImpl(CarRepository repository, CarRepository carRepository) {
         this.repository = repository;
     }
 
+    public Car findCarByModel() {
+        Car carToUpdate = repository.findCarByModel("IS-250");
+        return carToUpdate;
+    }
+
     @Override
-    public List<Car> getCars() {
+    public List<Car> getAll() {
         return repository.findAll();
     }
+
+    @Override
+    public Car update() {
+        Car carToUpdate = findCarByModel();
```

Comment:
carToUpdate - wrong name. It would be better to use just 'car'. We have only one Car object in this method. Why not.